packages/preact/src/index.ts

```diff
@@ -320,38 +320,28 @@ Component.prototype.shouldComponentUpdate = function (
 	const updater = this._updater;
 	const hasSignals = updater && updater._sources !== undefined;
 
-	// let reason;
-	// if (!hasSignals && !hasComputeds.has(this)) {
-	// 	reason = "no signals or computeds";
-	// } else if (hasPendingUpdate.has(this)) {
-	// 	reason = "has pending update";
-	// } else if (hasHookState.has(this)) {
-	// 	reason = "has hook state";
-	// }
-	// if (reason) {
-	// 	if (!this) reason += " (`this` bug)";
-	// 	console.log("not optimizing", this?.constructor?.name, ": ", reason, {
-	// 		details: {
-	// 			hasSignals,
-	// 			hasComputeds: hasComputeds.has(this),
-	// 			hasPendingUpdate: hasPendingUpdate.has(this),
-	// 			hasHookState: hasHookState.has(this),
-	// 			deps: Array.from(updater._deps),
-	// 			updater,
-	// 		},
-	// 	});
-	// }
-
-	// if this component used no signals or computeds, update:
-	if (!hasSignals && !(this._updateFlags & HAS_COMPUTEDS)) return true;
-
-	// if there is a pending re-render triggered from Signals,
-	// or if there is hook or class state, update:
-	if (this._updateFlags & (HAS_PENDING_UPDATE | HAS_HOOK_STATE)) return true;
-
+	// If this is a component using state, rerender
 	// @ts-ignore
 	for (let i in state) return true;
 
+	if (this.__f || (typeof this.u == "boolean" && this.u === true)) {
```

Comment:
I'd love to have opinions on this retroactive fix, with adding `this.u` here we don't necessarily do anything wrong, when hooks would settle into a falsey update they would never call this function before so as of 10.25.x it fixes an issue and in pre 10.24.x it doesn't really do anything